docs/visual-testing/integrations/java.md

```diff
@@ -0,0 +1,166 @@
+
+### Step 1: Add Sauce Labs Visual dependecy
+
+Add [Sauce Visual](https://central.sonatype.com/artifact/com.saucelabs.visual/java-client) dependency to your pom.xml
+  ```xml
+  <dependency>
+    <groupId>com.saucelabs.visual</groupId>
+    <artifactId>java-client</artifactId>
+    <version>0.3.107</version>
+    <scope>test</scope>
+  </dependency>
+  ```
+
+*Note: You can find the latest versions available [here](https://central.sonatype.com/search?q=com.saucelabs.visual).*
+
+### Step 2: Configure Visual Testing integration
+
+Declare a RemoteWebDriver and a VisualApi instance as class variables
+```java
+  private static VisualApi visual;
+  private static RemoteWebDriver driver;
+```
+
+Initialize `RemoteWebDriver` and `VisualApi` in `@BeforeSuite` section
+```java
+  @BeforeSuite
+  public static void init() {
+      driver = new RemoteWebDriver(webDriverUrl, capabilities);
+      visual = new Builder(driver, sauceUsername, sauceAccessKey, DataCenter.US_WEST_1).build();
+  }
+```
+
+Don't forget to quit the WebDriver in `@AfterSuite` section
+```java
+  @AfterSuite
+      public static void tearDown() {
+          if (driver != null) {
+              driver.quit();
+      }
+  }
+```
+
+
+### Step 3: Add visual tests in your tests
+
+Add a check to one of your tests:
+```java
+    @Test
+    void checkLoginLooksTheSame() {
+        var loginPage = new LoginPage(driver);
+        loginPage.open();
+
+        visual.check("Before Login");
+    }
+```
+
+### Step 4: Configure your Sauce Labs credentials
+
+Sauce Labs Visual relies on environment variables for authentications.<br />
+Both `SAUCE_USERNAME` and `SAUCE_ACCESS_KEY` need to be set prior starting your WebdriverIO job.
+
+Username and Access Key can be retrieved from https://app.saucelabs.com/user-settings.
+
+```sh
+export SAUCE_USERNAME=__YOUR_SAUCE_USER_NAME__
+export SAUCE_ACCESS_KEY=__YOUR_SAUCE_ACCESS_KEY__
+```
+
+### Step 5: Run the test
+
+Everything is now configured, your tests can be run as any other WebDriver test.
+
+Builds will appear on Sauce Labs platform as soon as they have been created by the running tests: https://app.saucelabs.com/visual/builds.
+
+## Advanced usage
+
+### Build attributes
+
+When creating the service in `VisualApi`, extra fields can be set to define the context, thus acting on which baselines new snapshots will be compared to. ([More info on baseline matching](../sauce-visual.md#baseline-matching))
+
+It needs to be defined through the `VisualApi.Builder` object.
+
+Methods available:
+- `withBuild(String build)`: Sets the name of the build
+- `withProject(String project)`: Sets the name of the project
+- `withBranch(String branch)`: Sets the name of the branch
+
+Example:
+```java
+    visual = new Builder(driver, username, accessKey, DataCenter.US_WEST_1)
+              .withBuild("Sauce Demo Test")
+              .withBranch("main")
+              .withProject("Java examples")
+              .build();
+```
+
+### Ignored regions
```

Comment:
Please always start with the component based ignore regions (that's the best practice) and the the coordinate